vhdbuilder/scripts/automate_version_bump.sh

```diff
@@ -21,6 +21,29 @@ find_and_write_build_timestamp() {
     echo "$json_string" > vhdbuilder/${NEW_IMAGE_VERSION}_build_timestamp.json
 }
 
+parse_and_write_base_image_version() {
+  # Use the VHD_BUILD_ID to go through publishing-info.json
+  # If 20.04 or higher, parse the base_image_version and append it to a json file
+  # Skip for others
+  base_image_version_json_file="vhdbuilder/${NEW_IMAGE_VERSION}_base_image_version.json"
+  if [ ! -f "$base_image_version_json_file" ]; then
+    echo "{}" > "$base_image_version_json_file"
+  fi
+
+  artifact_name=$(az pipelines runs artifact list --run-id ${VHD_BUILD_ID} | jq -r '.[].name' | grep "publishing-info" | awk '/2004|2204/')
+  artifacts=()
+  while IFS= read -r line; do
+    artifacts+=("$line")
+  done <<< "$artifact_name"
+  for artifact in "${artifacts[@]}"; do
+    az pipelines runs artifact download --artifact-name $artifact --path $(pwd) --run-id ${VHD_BUILD_ID}
+    BASE_IMAGE_VERSION=$(jq -r .base_image_version < vhd-publishing-info.json)
+    SKU_NAME=$(jq -r .sku_name < vhd-publishing-info.json)
+    jq --arg base_image_version "${BASE_IMAGE_VERSION}" --arg sku_name "${SKU_NAME}" '.[$sku_name] = $base_image_version' "$base_image_version_json_file" > tmp.json && mv tmp.json "$base_image_version_json_file"
```

Comment:
what exactly is this doing? are we creating a separate base image version file per SKU?

Reply:
answered offline

Reply:
explained offline, example commit: https://github.com/Azure/AgentBaker/commit/1ed95e75f0f218b7bd6af71329fb6d46a4b7efbb